### PR TITLE
LibGfx+LibGL+Tests: Test LibGL by comparing with QOI reference images

### PR DESCRIPTION
**Pull request by GMTA**
The introduction of `QOIWriter` was a good excuse to implement visual comparisons for the LibGL test suite. :^)

libgl-reference-images

### CODE REVIEW

**Comment by GMTA** [comment]
Marking this as a draft until the tests are also working in Lagom

File: Tests/LibGL/TestRender.cpp [hunk 1]
```diff
@@ -1,60 +1,100 @@
...
 {
-    auto bitmap = MUST(Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRx8888, { RENDER_WIDTH, RENDER_HEIGHT }));
+    auto bitmap = MUST(Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRx8888, { width, height }));
     auto context = GL::create_context(*bitmap);
-
     GL::make_context_current(context);
+    return context;
+}
+
+static void expect_bitmap_equals_reference(NonnullOwnPtr<GL::GLContext> context, StringView test_name)
```
Comment (sunverwerth) [inline Tests/LibGL/TestRender.cpp:32]: Why don't we simply pass a bitmap as first argument?

File: Tests/LibGL/TestRender.cpp [hunk 2]
```diff
@@ -1,60 +1,100 @@
...
-        // output the image to manually verify that the output is correct
-        Gfx::BMPWriter writer {};
-        auto buffer = writer.dump(bitmap);
-        int fd = open("./picture.bmp", O_CREAT | O_WRONLY, 0755);
-        EXPECT(fd > 0);
-        ssize_t nwritten = write(fd, buffer.data(), buffer.size());
-        EXPECT_EQ((size_t)nwritten, buffer.size());
-        close(fd);
-    }
+    expect_bitmap_equals_reference(move(context), "0002_quad_color_interpolation");
```
Comment (sunverwerth) [inline Tests/LibGL/TestRender.cpp:99]: See above. Why not `expect_bitmap_equals_reference(context->frontbuffer(), "0002_quad_color_interpolation");` ?

**Review by sunverwerth — COMMENTED** [review]
Yeah looks good overall. Left 2 remarks about the general context/bitmap handling but that's personal preference I guess.

File: Tests/LibGL/TestRender.cpp (hunk 1 above)
Comment (GMTA) [reply Tests/LibGL/TestRender.cpp]: I disliked having `context->present()` in every test, but needing to `move()` the context into the function doesn't make a lot of sense in hindsight. Changed it so it accepts a `Gfx::Bitmap` instead!

**Review by sunverwerth — APPROVED** [review]

**Comment by sunverwerth** [comment]
LGTM

**Pull request merged**